software/base/src/main/java/brooklyn/entity/software/MachineLifecycleEffectorTasks.java

```diff
@@ -151,8 +151,18 @@ public void attachLifecycleEffectors(Entity entity) {
                 .build();
     }
 
+    /** @see {@link #newStartEffector()} */
+    public Effector<Void> newSuspendEffector() {
+        return Effectors.effector(Void.class, "suspend")
+                .description("Suspend the process/service represented by an entity")
+                .parameter(StopSoftwareParameters.STOP_PROCESS_MODE)
+                .parameter(StopSoftwareParameters.STOP_MACHINE_MODE)
+                .impl(newSuspendEffectorTask())
+                .build();
+    }
```

Comment:
Suspending a machine has the same semantics as stopping a machine. They both use `MachineLifecycleEffectorTasks#doStop`. The former calls `MachineProvisioningLocation#release` (which becomes `JcloudsLocation#releaseNode`), the latter `MachineManagementMixins.SuspendsMachines#suspendMachine` (which becomes `JcloudsLocation#suspendMachine`). A suspended machine cannot subsequently be stopped if `IF_NOT_STOPPED` is true because `MachineLifecycleEffectorTasks#doStop` sets the entity's state to `STOPPED`.